src/compiler/transformers/module/es6.ts

```diff
@@ -4,9 +4,35 @@
 /*@internal*/
 namespace ts {
     export function transformES6Module(context: TransformationContext) {
+        const compilerOptions = context.getCompilerOptions();
+        const resolver = context.getEmitResolver();
+
+        let currentSourceFile: SourceFile;
+
         return transformSourceFile;
 
         function transformSourceFile(node: SourceFile) {
+            if (isExternalModule(node) || compilerOptions.isolatedModules) {
+                currentSourceFile = node;
+                return visitEachChild(node, visitor, context);
+            }
+            return node;
+        }
+
+        function visitor(node: Node) {
+            switch (node.kind) {

```

Comment:
I used a switch as I'm pretty sure I will need to circle back to handle export elision with an ExportDeclaration but I'm waiting until I run into that.